test/find_j_moves_test.cpp

```diff
@@ -0,0 +1,166 @@
+#include <map>
+#include <algorithm>
+#include <initializer_list>
+#include <gtest/gtest.h>
+#include <wl/game_state.h>
+#include <wl/find_j_moves.h>
+#include <wl/default_map.h>
+
+namespace wl {
+std::ostream& operator<<(std::ostream& os, JackResource resource) {
+	switch (resource) {
+	case JackResource::CARRIAGE: os << "CARRIAGE"; break;
+	case JackResource::ALLEY: os << "ALLEY"; break;
+	case JackResource::BOAT: os << "BOAT"; break;
+	}
+	return os;
+}
+
+std::ostream& operator<<(std::ostream& os, JackMove move) {
+	os << "JackMove{destination:" << static_cast<size_t>(move.destination());
+	if (const auto cost = move.cost(); cost.has_value()) {
+		os << ",cost:" << *cost;
+	}
+	os << "}";
+	return os;
+}
+} // namespace wl
```

Comment:
For the purposes of testing, I tell the command-line printing code how to pretty-print an instance of `wl::JackMove`.

Should look like either:

```
JackMove{destination:123}
```
or 
```
JackMove{destination:123,cost:CARRIAGE}
```